sdk/program/src/epoch_rewards_partition_list.rs

```diff
@@ -0,0 +1,36 @@
+use {
+    crate::{hash::Hash, pubkey::Pubkey},
+    serde_derive::{Deserialize, Serialize},
+};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub enum EpochRewardsPartitionListVersion {
+    V0(EpochRewardsPartitionList),
+}
+
+/// List of partition data for the seed epoch
+pub type EpochRewardsPartitionList = Vec<PartitionData>;
+
+/// Data about a partition of epoch rewards
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub struct PartitionData {
+    /// Total rewards in this partition, in lamports
+    pub lamports: u64,
+    /// Hash of the list of `[(address, lamports), ...]` for this partition for
+    /// verification purposes
+    pub hash: Hash,
+}
+
+impl From<(u64, Hash)> for PartitionData {
+    fn from((lamports, hash): (u64, Hash)) -> Self {
+        Self { lamports, hash }
+    }
+}
+
+pub fn get_epoch_rewards_partition_list_address(epoch: u64) -> Pubkey {
+    let (address, _bump_seed) = Pubkey::find_program_address(
+        &[b"EpochRewardsPartitionList", &epoch.to_le_bytes()],
```

Comment:
We can certainly do that. It will be somewhat annoying for users to have to look up that previous blockhash in order to derive this address.

Reply:
yeah, would probably just stick it in the rewards period metadata account too.  kinda breaks the ledger-only historical lookup tho :thinking:

Reply:
Handling the previous blockhash is actually nicer in the ledger-only case, because we store `parent_blockhash` in block metadata.
In this PDA case, I do like the idea of storing it in the account. But I haven't come up with a reason it should be part of the address seeds. Only the stake program can write to the account, so I don't think the account can be griefed, can it?

Reply:
> i _vaguely_ remember wanting to seed this address with the blockhash of the last block in the previous epoch. need to recover my rational. something something anti-griefing

for posterity, i was misremembering protecting this pda from griefing for protecting the partition hasher initialization. this is fine. we're already protecting the hasher